Assets/Scripts/UI/InGame/HUD/ChangeHUDHandler.cs

```diff
@@ -0,0 +1,143 @@
+﻿using UnityEngine;
+using UnityEngine.UI;
+using TMPro;
+using Assets.Scripts.UI.InGame.HUD;
+
+namespace Assets.Scripts.UI.InGame
+{
+    public class ChangeHudHandler : UiMenu
+    {
+        public InGameMenu menu;
+        public HUD.HUD HUD;
+        public GameObject[] HUDelements;
+        public bool elementSelected = false;
+        public GameObject selectedElement;
+        public Slider scaleSlider;
+        public TMP_Text elementLabel;
+        public Toggle toggleVisibility;
+        public bool hasChanged = false;
+
+        protected override void OnEnable()
+        {
+            base.OnEnable();
+        }
+
+        protected override void OnDisable() {
+            base.OnDisable();
+		}
+
+
+        public void Update()
+        {
+            if(HUD.inEditMode)
+            {
+                if(selectedElement != null)
+                {
+                    elementLabel.text = selectedElement.GetComponent<CustomizableHudElement>().elementName;
+                    selectedElement.transform.localScale = new Vector3(scaleSlider.value, scaleSlider.value, 1);
+                    selectedElement.GetComponent<CustomizableHudElement>().isVisible = toggleVisibility.isOn;
+                } else
+                {
+                    elementLabel.text = "Select an Element";
+                }
+            } else
+            {
+                SetVisibility();
+            }
+        }
+
+        public void SetVisibility()
+        {
+            foreach(GameObject element in HUDelements)
+            {
+                element.GetComponent<CustomizableHudElement>().SetVisibility();
+            }
+        }
+
+        // Called when the user starts editing the HUD after clicking the "Change HUD" button.
+        public void EnterEditMode()
+        {
+            HUD.inEditMode = true;
+
+            foreach(GameObject element in HUDelements)
```

Comment:
Now that HUDElements if of type CustomizableHudElement, you don't have to call all those ugly GetComponent<> anymore. If you still had to, then it should only be used once, and then stored as a variable.